app/api/v1/auth/index.js

```diff
@@ -82,16 +83,69 @@ router.post('/register', (req, res, next) => {
     return next(new error.BadRequest('Password must be at least 10 characters long'));
   }
 
-  // TODO account activation via email
+
   const newUser = User.sanitize(req.body.user);
   newUser.state = 'ACTIVE';
   User.generateHash(req.body.user.password).then((hash) => {
     newUser.hash = hash;
-    return User.create(newUser);
-  }).then((user) => {
-    log.info('user authentication (registration)', { request_id: req.id, user_uuid: user.uuid });
-    res.json({ error: null, user: user.getPublicProfile() });
-    Activity.accountCreated(user.uuid);
+    // require that we create a user succesfully and send email succesfully
+    db.transaction({
+      isolationLevel: Sequelize.Transaction.ISOLATION_LEVELS.REPEATABLE_READ,
+    }, (transaction) => User.generateAccessCode().then((code) => {
+      newUser.accessCode = code;
+      return User.create(newUser).then((user) => {
+        email.sendEmailVerification(user.email, user.firstName, code).then(() => {
+        }).catch(() => {
+          throw new error.BadRequest(`Something went wrong with sending email verification to ${user.email}`);
+        });
+        return user;
+      });
+    })).then((transactRes) => {
+      const user = transactRes;
+      log.info('user authentication (registration)', { request_id: req.id, user_uuid: user.uuid });
+      res.json({ error: null, user: user.getPublicProfile() });
+      Activity.accountCreated(user.uuid);
+    }).catch((err) => next(err));
+  }).catch(next);
+});
+
+/**
+ * Emails the user a email verification link
+ */
+// TODO: RATE LIMIT THIS
+router.get('/emailVerification/:email', (req, res, next) => {
+  if (!req.params.email) return next(new error.BadRequest('Email must be provided!'));
+  User.findByEmail(req.params.email).then((user) => {
+    if (!user) throw new error.BadRequest('Invalid email');
+    User.generateAccessCode().then(async (code) => {
+      user.accessCode = code;
+      // wait for user to save first before sending email
+      await user.save();
+      email.sendEmailVerification(user.email, user.firstName, code).then(() => {
+        res.json({ error: null });
+      }).catch(next);
+    });
+  }).catch(next);
+});
+
+/**
+ * Verifies a user's email based on a code sent out to the user's provided email
+ * Responds with a JSON object with fields 'error', 'verified' (if a user is succesfully verified)
+ */
+router.post('/emailVerification/:accessCode', (req, res, next) => {
+  if (!req.params.accessCode) {
+    return next(new error.BadRequest('Email verification code must be provided'));
+  }
+  User.findByAccessCode(req.params.accessCode).then(async (user) => {
+    if (!user) throw new error.BadRequest('Code is invalid');
+    if (user.accessCode === req.params.accessCode) {
```

Comment:
Isn't this implied by `User.findByAccessCode`? Like, `user` exists only if the access code is a match so this conditional isn't doing anything.